scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py

```diff
@@ -0,0 +1,243 @@
+        self.out_files_root_dir = path.join(path.expanduser("~"), "Engineering_Mantid/")
+
+    def create_new_calibration(self,
+                               vanadium_path,
+                               ceria_path,
+                               plot_output,
+                               instrument,
+                               rb_num=None):
+        """
+        Create a new calibration from a vanadium run and ceria run
+        :param vanadium_path: Path to vanadium data file.
+        :param ceria_path: Path to ceria data file
+        :param plot_output: Whether the output should be plotted.
+        :param instrument: The instrument the data relates to.
+        :param rb_num: The RB number for file creation.
+        """
+
+        vanadium_corrections = self.calculate_vanadium_correction(vanadium_path)
+        van_integration = vanadium_corrections[0]
+        van_curves = vanadium_corrections[1]
+        ceria_workspace = self.load_ceria(ceria_path)
+        output = self.run_calibration(ceria_workspace, van_integration, van_curves)
+        if plot_output:
+            self._plot_vanadium_curves()
+            for i in range(2):
+                difc = [output[i].DIFC]
+                tzero = [output[i].TZERO]
+                self._plot_difc_zero(difc, tzero)
+        difc = [output[0].DIFC, output[1].DIFC]
+        tzero = [output[0].TZERO, output[1].TZERO]
+
+        calibration_dir = self.out_files_root_dir + "Calibration/"
+        self.create_output_files(calibration_dir, difc, tzero, ceria_path, vanadium_path,
+                                 instrument)
+        if rb_num:
+            calibration_dir = self.out_files_root_dir + "User/" + rb_num + "/Calibration/"
+            self.create_output_files(calibration_dir, difc, tzero, ceria_path, vanadium_path,
+                                     instrument)
+
+    @staticmethod
+    def _plot_vanadium_curves():
+        van_curve_twin_ws = "__engggui_vanadium_curves_twin_ws"
+
+        if Ads.doesExist(van_curve_twin_ws):
+            DeleteWorkspace(van_curve_twin_ws)
+        CloneWorkspace(InputWorkspace="engggui_vanadium_curves", OutputWorkspace=van_curve_twin_ws)
+        van_curves_ws = Ads.retrieve(van_curve_twin_ws)
+        for i in range(1, 3):
+            if i == 1:
+                curve_plot_bank_1 = plot([van_curves_ws], [0, 1, 2])
+                curve_plot_bank_1.gca().set_title("Engg GUI Vanadium Curves Bank 1")
+                curve_plot_bank_1.gca().legend(["Data", "Calc", "Diff"])
+            if i == 2:
+                curve_plot_bank_2 = plot([van_curves_ws], [3, 4, 5])
+                curve_plot_bank_2.gca().set_title("Engg GUI Vanadium Curves Bank 2")
+                curve_plot_bank_2.gca().legend(["Data", "Calc", "Diff"])
+
+    @staticmethod
+    def _plot_difc_zero(difc, tzero):
+        for i in range(1, 3):
+            bank_ws = Ads.retrieve("engggui_calibration_bank_" + str(i))
```

Comment:
Can we use the `_generate_table_workspace_name` function to avoid repeating the prefix?